backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/RunVmCommand.java

```diff
@@ -956,10 +958,21 @@ protected boolean getVdsToRunOn() {
         }
 
         addCpuAndNumaPinning();
+        setDedicatedCpus();
 
         return true;
     }
 
+    private void setDedicatedCpus() {
+        if (StringUtils.isNotEmpty(getVm().getCurrentCpuPinning())) {
+            // Auto pinning in action, we don't wish to override.
+            return;
+        }
+        List<VdsCpuUnit> vdsCpuUnits = resourceManager.getVdsManager(getVdsId()).getCpuTopology().stream().filter(cpu -> cpu.getVmId() != null && cpu.getVmId().equals(getVmId())).collect(Collectors.toList());
```

Comment:
you can replace `cpu.getVmId() != null && cpu.getVmId().equals(getVmId())`  with `getVmId().equals(cpu.getVmId())`